File: src/main/prim/etc/_loop.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.nvm.{ AssemblerAssistant, Command, Context, CustomAssembled }

class _loop extends Command with CustomAssembled {
  override def perform(context: Context) {
    // we get custom-assembled out of existence
    throw new IllegalStateException()
  }
  override def assemble(a: AssemblerAssistant) {
    a.comeFrom()
    a.block()
    a.goTo()
  }
}

File: netlogo-gui/src/main/prim/etc/_loop.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.core.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;

public final strictfp class _loop
    extends Command
    implements org.nlogo.nvm.CustomAssembled {

  @Override
  public void perform(final Context context) {
    // we get custom-assembled out of existence
    throw new IllegalStateException();
  }

  public void assemble(org.nlogo.nvm.AssemblerAssistant a) {
    a.comeFrom();
    a.block();
    a.goTo();
  }
}
